miio/fan_miot.py

```diff
@@ -525,3 +547,312 @@ def delay_off(self, minutes: int):
+            "Oscillate: {result.oscillate}\n"
+            "Power: {result.power}\n"
+            "Powersupply Attached: {result.powersupply_attached}\n"
+            "Speed RPM: {result.speed_rpm}\n"
+            "Temperature: {result.temperature}\n",
+        )
+    )
+    def status(self):
+        """Retrieve properties."""
+        return FanStatusZA5(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    @command(default_output=format_output("Powering on"))
+    def on(self):
+        """Power on."""
+        return self.set_property("power", True)
+
+    @command(default_output=format_output("Powering off"))
+    def off(self):
+        """Power off."""
+        return self.set_property("power", False)
+
+    @command(default_output=format_output("Turning ionizer on"))
+    def ionizer_on(self):
+        """Turn ionizer on."""
+        return self.set_property("anion", True)
+
+    @command(default_output=format_output("Turning ionizer off"))
+    def ionizer_off(self):
+        """Turn ionizer off."""
+        return self.set_property("anion", False)
+
+    @command(
+        click.argument("speed", type=int),
+        default_output=format_output("Setting speed to {speed}%"),
+    )
+    def set_speed(self, speed: int):
+        """Set fan speed."""
+        if speed < 1 or speed > 100:
+            raise FanException("Invalid speed: %s" % speed)
+
+        return self.set_property("fan_speed", speed)
+
+    @command(
+        click.argument("angle", type=int),
+        default_output=format_output("Setting angle to {angle}"),
+    )
+    def set_angle(self, angle: int):
+        """Set the oscillation angle."""
+        if angle not in SUPPORTED_ANGLES[self.model]:
+            raise FanException(
+                "Unsupported angle. Supported values: "
+                + ", ".join("{0}".format(i) for i in SUPPORTED_ANGLES[self.model])
+            )
+
+        return self.set_property("swing_mode_angle", angle)
+
+    @command(
+        click.argument("oscillate", type=bool),
+        default_output=format_output(
+            lambda oscillate: "Turning on oscillate"
+            if oscillate
+            else "Turning off oscillate"
+        ),
+    )
+    def set_oscillate(self, oscillate: bool):
+        """Set oscillate on/off."""
+        if oscillate:
+            return self.set_property("swing_mode", True)
+        else:
+            return self.set_property("swing_mode", False)
+
+    @command(
+        click.argument("buzzer", type=bool),
+        default_output=format_output(
+            lambda buzzer: "Turning on buzzer" if buzzer else "Turning off buzzer"
+        ),
+    )
+    def set_buzzer(self, buzzer: bool):
+        """Set buzzer on/off."""
+        if buzzer:
+            return self.set_property("buzzer", True)
+        else:
+            return self.set_property("buzzer", False)
```

Comment:
```suggestion
        return self.set_property("buzzer", buzzer)
```

Reply:
Done. This code was copy-pasted from `FanMiot`, so I have made the fixes there as well.

Reply:
Ah, thanks! Maybe there should be a generic base class for all those fans, but that can wait for now.